### PR TITLE
Create weekly-update.2015-04-03.md

### PR DESCRIPTION
**Pull request by Gioyik**

### CODE REVIEW

File: weekly-updates/weekly-update.2015-04-03.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+* **iojs**: an "internal modules" API has been introduced to allow core code to share JavaScript modules internally only without having to expose them as a public API, this feature is for core-only [#848](https://github.com/iojs/io.js/pull/848) (Vladimir Kurchatkin).
+* **timers**: two minor problems with timers have been fixed:
+  - `Timer#close()` is now properly idempotent [#1288](https://github.com/iojs/io.js/issues/1288) (Petka Antonov).
+  - `setTimeout()` will only run the callback once now after an `unref()` during the callback [#1231](https://github.com/iojs/io.js/pull/1231) (Roman Reiss).
+  - NOTE: there are still other unresolved concerns with the timers code, such as [#1152](https://github.com/iojs/io.js/pull/1152).
+* **Windows**: a "delay-load hook" has been added for compiled add-ons on Windows that should alleviate some of the problems that Windows users may be experiencing with add-ons in io.js [#1251](https://github.com/iojs/io.js/pull/1251) (Bert Belder).
+* **V8**: minor bug-fix upgrade for V8 to 4.1.0.27.
+* **npm**: upgrade npm to 2.7.4. See [npm CHANGELOG.md](https://github.com/npm/npm/blob/master/CHANGELOG.md#v274-2015-03-20) for details. Summary:
```
Comment (yosuke-furukawa) [inline weekly-updates/weekly-update.2015-04-03.md]: We don't write npm change log in detail. 
https://github.com/iojs/evangelism/pull/37/files
https://github.com/iojs/evangelism/issues/16#issuecomment-77664152

File: weekly-updates/weekly-update.2015-04-03.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+* Some problems exist with timers and `unref()` still to be resolved. See [#1152](https://github.com/iojs/io.js/pull/1152).
+* Possible small memory leak(s) may still exist but have yet to be properly identified, details at [#1075](https://github.com/iojs/io.js/issues/1075).
+* Surrogate pair in REPL can freeze terminal [#690](https://github.com/iojs/io.js/issues/690)
+* Not possible to build io.js as a static library [#686](https://github.com/iojs/io.js/issues/686)
+* `process.send()` is not synchronous as the docs suggest, a regression introduced in 1.0.2, see [#760](https://github.com/iojs/io.js/issues/760) and fix in [#774](https://github.com/iojs/io.js/issues/774)
+* Calling `dns.setServers()` while a DNS query is in progress can cause the process to crash on a failed assertion [#894](https://github.com/iojs/io.js/issues/894)
+
+### Commits
+
```
Comment (yosuke-furukawa) [inline weekly-updates/weekly-update.2015-04-03.md]: We don't write commit log. please remove them.

**Comment by Gioyik** [comment]
Changes requested by @yosuke-furukawa are done. Let me know if there's something to change/add.

File: weekly-updates/weekly-update.2015-04-03.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* Calling `dns.setServers()` while a DNS query is in progress can cause the process to crash on a failed assertion [#894](https://github.com/iojs/io.js/issues/894)
+
+# Community Updates
+
+* [Scaleway](https://www.scaleway.com/) provides some ARM resources for the iojs test/build infrastructure.
+* New post on Medium about Node.js reconciliation: [Help us reconcile node.js and io.js](https://medium.com/node-js-javascript/help-us-reconcile-node-js-and-io-js-c060a9ec1bd4)
+* Added support for iojs in [Reactive-Extensions/RxJS](https://travis-ci.org/Reactive-Extensions/RxJS/builds/56671837)
+* [joyent/nodejs-advisory-board#30](https://github.com/joyent/nodejs-advisory-board/pull/30) merged
+* Mikeal Rogers working on reconciling Project Lifecyce and WIG [joyent/nodejs-advisory-board#33](https://github.com/joyent/nodejs-advisory-board/pull/33)
+
```
Comment (yosuke-furukawa) [inline weekly-updates/weekly-update.2015-04-03.md:33]: Please add this news for every iojs devs!! 
https://github.com/iojs/evangelism/issues/51#issuecomment-89493762

**Comment by yosuke-furukawa** [comment]
I will merge if you add the news.

**Comment by Gioyik** [comment]
@yosuke-furukawa done :)

**Comment by diagramatics** [comment]
Formatting nitpicking. Community updates and upcoming events should be on heading level 3 as well as the title.

**Comment by Gioyik** [comment]
@diagramatics I created this using the template used before for weekly updates: 

https://github.com/iojs/evangelism/blob/master/weekly-updates/weekly-update.2015-03-20.md

**Comment by diagramatics** [comment]
It should be on level 3 for easier copy and paste to Medium's editor without having to reformat the text again. I mentioned about this earlier but it seems the change was only made in the Medium posts and not in the Markdown files as well.

File: weekly-updates/weekly-update.2015-04-03.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+* `process.send()` is not synchronous as the docs suggest, a regression introduced in 1.0.2, see [#760](https://github.com/iojs/io.js/issues/760) and fix in [#774](https://github.com/iojs/io.js/issues/774)
+* Calling `dns.setServers()` while a DNS query is in progress can cause the process to crash on a failed assertion [#894](https://github.com/iojs/io.js/issues/894)
+
+# Community Updates
+
+* [Scaleway](https://www.scaleway.com/) provides some ARM resources for the iojs test/build infrastructure.
+* New post on Medium about Node.js reconciliation: [Help us reconcile node.js and io.js](https://medium.com/node-js-javascript/help-us-reconcile-node-js-and-io-js-c060a9ec1bd4)
+* Added support for iojs in [Reactive-Extensions/RxJS](https://travis-ci.org/Reactive-Extensions/RxJS/builds/56671837)
+* [joyent/nodejs-advisory-board#30](https://github.com/joyent/nodejs-advisory-board/pull/30) merged
+* Mikeal Rogers working on reconciling Project Lifecyce and WIG [joyent/nodejs-advisory-board#33](https://github.com/joyent/nodejs-advisory-board/pull/33)
```
Comment (thefourtheye) [inline weekly-updates/weekly-update.2015-04-03.md]: Lifecycle, typo

Comment (yosuke-furukawa) [inline weekly-updates/weekly-update.2015-04-03.md]: WIG is WG??

File: weekly-updates/weekly-update.2015-04-03.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+This week we had one io.js release [v1.6.3](https://iojs.org/dist/v1.6.3/), complete changelog can be found [on GitHub](https://github.com/iojs/io.js/blob/v1.x/CHANGELOG.md).
+
+### Notable changes
+
+#### 1.6.3
+
+* **fs**: corruption can be caused by `fs.writeFileSync()` and append-mode `fs.writeFile()` and `fs.writeFileSync()` under certain circumstances, reported in [#1058](https://github.com/iojs/io.js/issues/1058), fixed in [#1063](https://github.com/iojs/io.js/pull/1063) (Olov Lassus).
+* **iojs**: an "internal modules" API has been introduced to allow core code to share JavaScript modules internally only without having to expose them as a public API, this feature is for core-only [#848](https://github.com/iojs/io.js/pull/848) (Vladimir Kurchatkin).
+* **timers**: two minor problems with timers have been fixed:
+  - `Timer#close()` is now properly idempotent [#1288](https://github.com/iojs/io.js/issues/1288) (Petka Antonov).
```
Comment (kosamari) [inline weekly-updates/weekly-update.2015-04-03.md]: Typo : `Timer.close()` ??

Comment (Gioyik) [inline weekly-updates/weekly-update.2015-04-03.md]: Yes, It's a typo problem on io.js changelog.

**Comment by Gioyik** [comment]
All suggested changes are done. Feel free to do a double check to see if this is ok to merge (let me know if you want this PR squashed).

I want to discuss something. Should we change in past weekly updates from h1 to h3 titles? or just keep them in that form because are already published?

Thanks

**Comment by yosuke-furukawa** [comment]
> Should we change in past weekly updates from h1 to h3 titles? 

I don't have strong opinion... I vote for keep the style, those update are already published.

**Comment by yosuke-furukawa** [comment]
LGTM

**Comment by yosuke-furukawa** [comment]
@iojs/evangelism please publish our weekly updates on medium.

**Comment by diagramatics** [comment]
The published version has its heading levels changed already before posted on Medium. It's just that the Markdown files weren't updated to match the new formatting.

File: weekly-updates/weekly-update.2015-04-03.md (hunk 5 above)
Comment (Fishrock123) [inline weekly-updates/weekly-update.2015-04-03.md]: `Timer#close()` may have been correct: https://github.com/iojs/io.js/pull/1342#issuecomment-89770128

Comment (Gioyik) [inline weekly-updates/weekly-update.2015-04-03.md]: This was not a typo issue:

https://github.com/iojs/io.js/pull/1342#issuecomment-89770128

**Pull request merged**